sample-app/src/components/DropdownSection.tsx

```diff
@@ -0,0 +1,108 @@
+import classNames from "classnames";
+import { useState, useEffect } from "react";
+
+export interface Option {
+  value: string,
+  display: JSX.Element
+}
+
+interface DropdownSectionProps<T> {
+  focusStatus?: string,
+  options: T[],
+  optionIdPrefix: string,
+  onFocusChange: (value: string, focusedOptionId: string) => void,
+  onLeaveSectionFocus?: (focusNext: boolean) => void,
+  onClickOption: (option: T, optionIndex: number) => void,
+  label?: string,
+  cssClasses: {
+    sectionContainer: string,
+    sectionLabel: string,
+    optionsContainer: string,
+    option: string,
+    focusedOption: string
+  }
+}
+
+export default function DropdownSection<T extends Option>({
+  focusStatus,
+  options,
+  optionIdPrefix,
+  onFocusChange,
+  onLeaveSectionFocus = () => {},
+  onClickOption,
+  label = '',
+  cssClasses
+}: DropdownSectionProps<T>): JSX.Element | null {
+
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState<number>(0);
+
+  function changeOptionFocus(focusNext: boolean) {
```

Comment:
What do you think of having `incrementOptionFocus` and `decrementOptionFocus` instead? I think it's a little clearer than the `focusNext` boolean flag

Reply:
agree with splitting into increment/decrement , I think `focusNext` boolean can be a bit confusing

Reply:
if I split it into 2 functions, it would look something like:
```
  function incrementOptionFocus() {
    let newIndex = focusedOptionIndex + 1;
    if (newIndex < options.length) {
      onFocusChange(options[newIndex].value, `${optionIdPrefix}-${newIndex}`);
    } else {
      onLeaveSectionFocus(true);
      newIndex = options.length - 1;
    }
    setFocusedOptionIndex(newIndex);
  }

  function decrementOptionFocus() {
    let newIndex = focusedOptionIndex - 1;
    if (newIndex > -1) {
      onFocusChange(options[newIndex].value, `${optionIdPrefix}-${newIndex}`);
    } else {
      onLeaveSectionFocus(false);
      newIndex = 0;
    }
    setFocusedOptionIndex(newIndex);
  }
```
it seems a little repetitive to me, but if it's more clear, then I can change it to this

Reply:
I think the increased clarity is worth the repetition (2 lines of code)

Reply:
ok, changed!